### PR TITLE
add `match.restart_on_next_match` settings

### PR DESCRIPTION
**Pull request by zaoqi**

### CODE REVIEW

**Comment by Thomas--S** [comment]
Could you explain the goal of this PR please?

**Comment by ClobberXD** [comment]
Greetings, and thanks for wanting to contribute to CTF.

However I don't see why you want to convert match restart into a setting. This functionality is currently used for restarting the server at the end of the current match via a chat-command. By making that local variable store the value of the setting, you're making the server restart at the end of every single match, and you're also breaking the chat-command's functionality.

I can see the benefit of restarting the server after every match (e.g. due to every single match possibly being insanely long), but please use a separate variable for this purpose, so that the chat-command doesn't break.

File: mods/ctf/ctf_match/chat.lua [hunk 1]
```diff
@@ -79,7 +85,9 @@ minetest.register_chatcommand("ctf_unqueue_restart", {
 
 ctf_match.register_on_new_match(function()
 	if restart_on_next_match then
-		minetest.chat_send_player(restart_on_next_match_by, "Shutting down now!")
+		if restart_on_next_match_by ~= nil then
+			minetest.chat_send_player(restart_on_next_match_by, "Shutting down now!")
+		fi
```
Comment (ClobberXD) [inline mods/ctf/ctf_match/chat.lua]:
```suggestion
		end
```

**Comment by MinetestSam** [comment]
> I can see the benefit of restarting the server after every match (e.g. due to every single match possibly being insanely long).

I don't think players would enjoy being booted off after every single match irrespective of the time duration.

**Comment by ClobberXD** [comment]
> I don't think players would enjoy being booted off after every single match irrespective of the time duration.

That's exactly why I agree with this being a setting, and it defaulting to `false` for the sanity of the players.

**Comment by rubenwardy** [comment]
Restarting the server every match isn't a good idea

**Pull request closed without merging**